File: 2023/Leetcode/problem167.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        result = []

        left = 0
        right = len(numbers) - 1
        while left < right:
            total = numbers[left] + numbers[right]
            if total > target:
                right -= 1
            elif total < target:
                left += 1
            else:
                result = [left + 1, right + 1]
                left += 1
                right -= 1

        return result

File: 2024/Leetcode/problem167.js
/**
 * @param {number[]} numbers
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(numbers, target) {
    let left = 0;
    let right = numbers.length - 1;

    while (numbers[left] + numbers[right] != target) {
        if (numbers[left] + numbers[right] < target) {
            left++;
        } else {
            right--;
        }
    }

    return [left + 1, right + 1];
};